src/linear_algebra/matrix.lean

```diff
@@ -813,6 +820,26 @@ begin
   apply h₂,
 end
 
+variables {R M : Type*} [comm_ring R] [add_comm_group M] [module R M]
+variables {b : n → M} (hb : is_basis R b)
+
+lemma to_lin_bijective [decidable_eq n] (A : matrix n n R) (hA : is_unit A.det) :
+  function.bijective (to_lin hb hb A) :=
+begin
+  refine function.bijective_iff_has_inverse.mpr ⟨to_lin hb hb A⁻¹, λ x, _, λ x, _⟩;
+    simp only [← linear_map.comp_apply, ← to_lin_mul,
+               matrix.nonsing_inv_mul _ hA, matrix.mul_nonsing_inv _ hA,
+               to_lin_one, linear_map.id_apply]
+end
```

Comment:
(This could obviously lead into a linear equiv between matrices with `is_unit (det A)` and linear equivs, but I'd prefer to postpone that to a new PR.)

Reply:
One already has that for monoid_homs, basically. Can one extend `units.map_equiv` to `units.map_linear_equiv`?
```
-- The general linear group, as a subtype of matrices
abbreviation GLₙ (n : ℕ) (α : Type*) [comm_ring α] : Type* := units (matrix (fin n) (fin n) α)

lemma GLₙ.is_unit_det {n : ℕ} (A : GLₙ n ℝ) : is_unit (matrix.det (A : MnR n)) :=
begin
  rw ←matrix.is_unit_iff_is_unit_det,
  exact is_unit_unit _
end

lemma GLₙ.det_ne_zero {n : ℕ} (A : GLₙ n ℝ) : matrix.det (A : MnR n) ≠ 0 :=
begin
  rw ←is_unit_iff_det_ne_zero,
  exact is_unit_unit _
end

example {n : ℕ} : (GLₙ n ℝ) ≃* linear_map.general_linear_group ℝ (fin n → ℝ) :=
units.map_equiv (@linear_map.to_matrix_alg_equiv' ℝ _ (fin n) _ _).symm
```

Reply:
I don't think it will work, because `units` does not have scalar multiplication by a ring `R`: `0` times a unit is not a unit.